kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java

```diff
@@ -8,6 +10,9 @@
   // Replace all non-letter, non-digit characters with underscore. Append underscore in front of
   // name if it does not begin with alphabet or underscore.
   public static String sanitizeName(String name) {
+    if (name == null) {
+      throw new ConnectException("Record keys cannot be null");
```

Comment:
This might not be a record.

Reply:
What do you mean?

Reply:
A couple thoughts here:

1. It'd be great if we could get confirmation on whether or not this is actually causing issues for anybody right now. I'm wondering if the only source of the NPE was the bug we've already addressed with null values (as opposed to the check here, which is for null keys).
2. If this is still something we need to check for (presumably because it's invalid data for BigQuery and will be rejected once we send it to BigQuery), then we should perform the check for null map keys regardless of whether field name sanitization is enabled or not (so probably in a different class altogether--maybe the record converter?).

Reply:
1. I don't know the full scope of issues this is causing the customer, luckily there is the "workaround" of not sanitizing. This is the only time I've seen this issue, but also this is my first time on cloud on call. From what I understand, the only source of NPE is probably from the null values causing an NPE in `collect`.
2. I agree, although it seems like it is not invalid data for BigQuery, it seems like it is just an issue with the usage of `collect`. What do you think? It might be a good idea to add this check for null map keys as well.

Reply:
I think that given that this is a hotfix, we should prioritize getting the customer unblocked. If we can say with confidence that the issue with their connector is coming solely from null values, then we can skip the check for null keys and revisit that later. If we can't, then we should add that check in this PR.

Reply:
Can't some of the values be nulls if the schema supports it? I think we are hitting NPE just because the Collectors.toMap doesn't support null values.

Reply:
Right, I think we're all on the same page about that. The question is about whether that's the only issue that needs to be addressed here, or if we also need to address the potential for null keys.

Reply:
Do you know if BigQuery supports null keys? I've never seen an NPE in `sanitize`, which is where I believe the NPE would occur in the case of null keys. You can't sanitize a key which results in a null, but you can pass in a null key to `sanitize`.

Reply:
I don't know, and I suspect that it doesn't. But I think it's more important to figure out what's blocking the customer right now and focus our efforts accordingly. If nobody's running into that right now then let's take out the check for null keys from this PR and merge as-is, then revisit the issue of null keys later.

Reply:
Works for me, updating the PR right now!

Reply:
Yeah I would love to skip throwing exception for null keys here and not throw exception for that.

Reply:
PR is updated!